src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/AvailabilityService.kt

```diff
@@ -0,0 +1,55 @@
+package uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service
+
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.model.request.AvailabilityRequest
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.model.response.AvailabilityResponse
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.repository.VideoAppointmentRepository
+
+@Service
+class AvailabilityService(
+  private val videoAppointmentRepository: VideoAppointmentRepository,
+  private val availabilityFinderService: AvailabilityFinderService,
+) {
+
+  companion object {
+    private val log = LoggerFactory.getLogger(this::class.java)
+  }
+
+  /**
+   * Assumptions:
+   *  - we consider ONLY video link bookings (not other appointment types from NOMIS or A&A in these locations)
+   *  - we consider ONLY bookings present in the BVLS service, not in A&A or NOMIS (i.e. all video link bookings)
+   *  - current booking journeys for BVLS allows only one person at one prison per booking.
+   *  - we consider ONLY the prison locations in the request for alternative times (we do not consider others - yet)
+   *  - the decision about whether a user is allowed to book into a locations is still the users own.
+   *  - we assume all appointments for this booking are on the same day and at the same prison
+   *
+   * To consider later:
+   *  - Get other VCC-enabled rooms at these prison(s) and offer them?
+   *  - Incorporate prison room decoration logic here, to find other rooms?
+   */
```

Comment:
Almost, but not quite. The existing service calls whereabouts-api to get ALL appointments for these locations - video link bookings + other types too. My question this morning was related to this - whether we need to consider other types of appointment.  Sounds like we do, so I'll need to enhance this will calls out to either prison API or A&A API to get the others. One for a follow-on PR.